From df101a2aecdf29dc8a3578518709b2f1bd2117f1 Mon Sep 17 00:00:00 2001
From: saikumaru <saikumaru@ami.com>
Date: Tue, 9 Sep 2025 12:13:50 +0530
Subject: [PATCH] Add skip-soft-reset config to spi-nor driver.

Let spi-nor driver can decide to skip-soft-reset or not in dts.

Signed-off-by: saikumaru <saikumaru@ami.com>
---
 drivers/mtd/spi-nor/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mtd/spi-nor/core.c b/drivers/mtd/spi-nor/core.c
index 7ecc70e9245e..79b1cb7b9c94 100644
--- a/drivers/mtd/spi-nor/core.c
+++ b/drivers/mtd/spi-nor/core.c
@@ -3795,6 +3795,9 @@ static int spi_nor_probe(struct spi_mem *spimem)
 	if (ret)
 		return ret;
 
+	if (device_property_read_bool(&spi->dev, "skip-soft-reset"))
+		nor->flags &= ~SNOR_F_SOFT_RESET;
+
 	spi_nor_debugfs_register(nor);
 
 	/*
-- 
2.47.1

